tf2_ros/src/static_transform_broadcaster_program.cpp

```diff
@@ -148,16 +130,8 @@ int main(int argc, char ** argv)
     return -1;
   }
 
-  // Checks: frames should not be the same.
-  if (msg.header.frame_id == msg.child_frame_id)
-  {
-    RCUTILS_LOG_FATAL("target_frame and source frame are the same (%s, %s) this cannot work",
-                      msg.header.frame_id.c_str(), msg.child_frame_id.c_str());
-    return 1;
+  if (node) {
```

Comment:
This `if` shouldn't be necessary; there isn't any way to get here where the node would be null.  So I'd just remove this.

Reply:
True. Consider it gone[